sprout/appliances/tasks.py

```diff
@@ -1627,6 +1628,12 @@ def appliance_yum_update(self, appliance_id):
     appliance.ipapp.update_rhel(reboot=False)
 
 
+@singleton_task()
+def appliance_set_hostname(appliance_id):
+    appliance = Appliance.objects.get(id=appliance_id)
```

Comment:
REQUIRED: I believe you need to to check whether this appliance is hosted on openstack before doing changes.

Reply:
done.